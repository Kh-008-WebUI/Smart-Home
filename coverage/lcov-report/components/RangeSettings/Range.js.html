<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/RangeSettings/Range.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/RangeSettings</a> Range.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.37% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>38/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.1% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>23/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.92% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
import PropTypes from 'prop-types';
import './RangeStyle.scss';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span>
&nbsp;
&nbsp;
export default class RangeSettings extends React.Component {
  constructor (props) {<span class="branch-1 cbranch-no" title="branch not covered" ></span>
    super(props);
&nbsp;
    let minValue, maxValue;
&nbsp;
    if (this.props.params) {
      minValue = this.props.params.minValue;
      maxValue = this.props.params.maxValue;
    }
&nbsp;
    this.state = {
      value: 0,
      params: {
        minValue: minValue || 0,
        maxValue: maxValue || 100
      }
    };
&nbsp;
    this.setMinValue = (e) =&gt; {
      const min = parseInt(e.target.value) || 0;
&nbsp;
      if (min &lt;= this.state.params.maxValue) {
        this.setState({
          params: {
            minValue: min,
            maxValue: this.state.params.maxValue
          }
        });
      } else {
        this.setState({
          params: {
            minValue: min,
            maxValue: min
          }
        });
        this.maxElement.value = min;
      }
    };
&nbsp;
    this.setMaxValue = (e) =&gt; {
      const max = parseInt(e.target.value) || 0;
&nbsp;
      if (max &gt;= this.state.params.minValue) {
        this.setState({
          params: {
            maxValue: max,
            minValue: this.state.params.minValue
          }
        });
      } else {
        this.setState({
          params: {
            maxValue: max,
            minValue: max
          }
        });
        this.minElement.value = max;
      }
    };
  }
&nbsp;
  componentDidMount () {
    if (typeof this.props.data !== 'undefined') {
      this.setState({
        value: this.props.data
      });
    }
  }
&nbsp;
  componentWillReceiveProps (nextProps) {
    this.setState({
      value: nextProps.data || 0
    });
  }
&nbsp;
  onChange = (e) =&gt; {
    this.setState({
      value: e.target.value
    });
&nbsp;
    if (typeof this.props.onTimerChange !== 'undefined') {
      this.props.onTimerChange(e);
    }
  }
&nbsp;
  setValue = (e) =&gt; {
    if (typeof this.props.setTimerValue !== 'undefined') {
      this.props.setTimerValue();
    } else {
      this.props.setItemValue(this.state.value,
      this.props.itemId,
      this.props.deviceId);
    }
  }
&nbsp;
  setParams = () =&gt; {
    const params = {
      minValue: this.state.params.minValue || 0,
      maxValue: this.state.params.maxValue || 100
    };
&nbsp;
    this.props.setParameters(
      this.props.itemId,
      params);
  }<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span></span>
&nbsp;
  render () {
    const { min, max } = this.props;
&nbsp;
    return (
      &lt;div className={ `${ this.props.styleName }` }&gt;
        {this.props.showMinMax ?
          &lt;div className="range-settings"&gt;
            &lt;p className="range-settings__title"&gt;
              Enter min and max values for setting
            &lt;/p&gt;
            &lt;div className="range-settings__group"&gt;
              &lt;div className="range-settings__val"&gt;
                &lt;input
                  type="number"
                  name="min"
                  placeholder="Enter min value"
                  onChange={ this.setMinValue }
                  ref={(input) =&gt; {<span class="fstat-no" title="function not covered" ></span>
                    this.minElement = input;
                  }}
                  defaultValue={ this.state.params.minValue }
                  onBlur={ this.setParams }/&gt;
              &lt;/div&gt;
              &lt;div className="range-settings__val"&gt;
                &lt;input
                  type="number"
                  name="max"
                  ref={(input) =&gt; {<span class="fstat-no" title="function not covered" ></span>
                    this.maxElement = input;
                  }}
                  placeholder="Enter max value"
                  onChange={ this.setMaxValue }
                  defaultValue={ this.state.params.maxValue }
                  onBlur={ this.setParams }/&gt;
              &lt;/div&gt;
            &lt;/div&gt;
          &lt;/div&gt; :
          null
        }
        {this.props.hideLabel ? n<span class="branch-0 cbranch-no" title="branch not covered" >ull :</span> &lt;p className='range_value'&gt;
          {this.state.value}&lt;/p&gt;}
        &lt;input
          type='range'
          onChange={ this.onChange }
          onMouseUp={ this.setValue }
          min={min || this.state.params.minValue}
          max={max || this.state.params.maxValue}
          step={1}
          value={ this.state.value }
        /&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
RangeSettings.propTypes = {
  styleName: PropTypes.string,
  itemId: PropTypes.number,
  max: PropTypes.number,
  min: PropTypes.number,
  setDevice: PropTypes.func,
  hideLabel: PropTypes.bool,
  setItemValue: PropTypes.func,
  data: PropTypes.any,
  onTimerChange:PropTypes.func,
  deviceId: PropTypes.string,
  setTimerValue: PropTypes.func,
  setParameters: PropTypes.func,
  showMinMax: PropTypes.bool,
  maxValue: PropTypes.number,
  minValue: PropTypes.number,
  params: PropTypes.object
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 25 2017 01:58:35 GMT+0300 (Финляндия (лето))
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
