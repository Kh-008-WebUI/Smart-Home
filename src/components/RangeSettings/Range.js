import React from 'react';
import PropTypes from 'prop-types';
import './RangeStyle.scss';


export default class RangeSettings extends React.Component {
  constructor (props) {
    super(props);

    let minValue, maxValue;

    if (this.props.params) {
      minValue = this.props.params.minValue;
      maxValue = this.props.params.maxValue;
    }

    this.state = {
      value: 0,
      params: {
        minValue: minValue || 0,
        maxValue: maxValue || 100
      }
    };

    this.setMinValue = (e) => {
      const min = parseInt(e.target.value) || '';

      if (min <= this.state.params.maxValue) {
        this.setState({
          params: {
            minValue: min,
            maxValue: this.state.params.maxValue
          }
        });
      } else {
        e.target.setCustomValidity(`Value shoul be less then ${maxValue}`);
        e.target.reportValidity();
      }
    };

    this.setMaxValue = (e) => {
      const max = parseInt(e.target.value) || '';

      if (max > this.state.params.minValue) {
        this.setState({
          params: {
            maxValue: max,
            minValue: this.state.params.minValue
          }
        });
      } else {
        e.target.setCustomValidity(`Value shoul be less then ${minValue}`);
        e.target.reportValidity();
      }
    };
  }

  componentDidMount () {
    if (typeof this.props.data !== 'undefined') {
      this.setState({
        value: this.props.data
      });
    }
  }

  componentWillReceiveProps (nextProps) {
    this.setState({
      value: nextProps.data || 0
    });
  }

  onChange = (e) => {
    this.setState({
      value: e.target.value
    });

    if (typeof this.props.onTimerChange !== 'undefined') {
      this.props.onTimerChange(e);
    }
  }

  setValue = (e) => {
    if (typeof this.props.setTimerValue !== 'undefined') {
      this.props.setTimerValue();
    } else {
      this.props.setItemValue(this.state.value,
      this.props.itemId,
      this.props.deviceId);
    }
  }

  setParams = () => {
    const params = {
      minValue: this.state.params.minValue || 0,
      maxValue: this.state.params.maxValue || 100
    };

    this.props.setParameters(
      this.props.itemId,
      params);
  }

  render () {
    const { min, max } = this.props;

    return (
      <div className={ `${ this.props.styleName }` }>
        {this.props.showMinMax ?
          <div className="range-settings">
            <p className="range-settings__title">
              Enter min and max values for setting
            </p>
            <div className="range-settings__group">
              <div className="range-settings__val">
                <input
                  type="number"
                  name="min"
                  placeholder="Enter min value"
                  onChange={ this.setMinValue }
                  value={ this.state.params.minValue }
                  onBlur={ this.setParams }/>
              </div>
              <div className="range-settings__val">
                <input
                  type="number"
                  name="max"
                  placeholder="Enter max value"
                  onChange={ this.setMaxValue }
                  value={ this.state.params.maxValue }
                  onBlur={ this.setParams }/>
              </div>
            </div>
          </div> :
          null
        }
        {this.props.hideLabel ? null : <p className='range_value'>
          {this.state.value}</p>}
        <input
          type='range'
          onChange={ this.onChange }
          onMouseUp={ this.setValue }
          min={min || this.state.params.minValue}
          max={max || this.state.params.maxValue}
          step={1}
          value={ this.state.value }
        />
      </div>
    );
  }
}

RangeSettings.propTypes = {
  styleName: PropTypes.string,
  itemId: PropTypes.number,
  max: PropTypes.number,
  min: PropTypes.number,
  setDevice: PropTypes.func,
  hideLabel: PropTypes.bool,
  setItemValue: PropTypes.func,
  data: PropTypes.any,
  onTimerChange:PropTypes.func,
  deviceId: PropTypes.string,
  setTimerValue: PropTypes.func,
  setParameters: PropTypes.func,
  showMinMax: PropTypes.bool,
  maxValue: PropTypes.number,
  minValue: PropTypes.number,
  params: PropTypes.object
};
