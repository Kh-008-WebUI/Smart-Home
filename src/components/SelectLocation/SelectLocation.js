import React from 'react';
import PropTypes from 'prop-types';

export default class SelectLocation extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      input: false,
      locationValue: ''
    };
  }
  showInputLocation = () => {
    this.setState({
      input: !this.state.input
    });
  };

  changeLocationValue = (e) => {
    const newLocationValue = e.target.value;

    this.setState({
      locationValue: newLocationValue
    });
  };
  addLocationValue = () => {
    this.props.addLocation(this.state.locationValue);
  }
  render () {
    return (
      <div className="Select-control">
        <div className="Select-value select-menu-container">
          <div className="select-menu-label">
            <span className="Select-value-label">{this.state.locationValue}</span>
            <i className="select-toggle fa fa-caret-down"
            onClick={ this.showInputLocation }></i>
          </div>
        </div>
        {this.state.input ?
        <div>
          <div>
            <input type="text"
              placeholder="Add new location"
              className="Select-input_add-location"
              onChange={this.changeLocationValue}/>
            <i className="fa fa-plus Select-input_add-location_icon"
              onClick={this.addLocationValue}></i>
          </div>
              <ul className="Select-menu">
                {this.props.locations.map((location, i) => {
                  return (
                    <li key={i} className="Select-option"
                    onClick={this.changeLocationValue}>
                      {location.value}
                      <i className="fa fa-trash Select-option__icon"></i>
                    </li>
                  );
                })
                }
            </ul>
        </div> : null }
      </div>
    );
  }
}
SelectLocation.propTypes = {
  locations: PropTypes.array,
  addLocation: PropTypes.func
};
