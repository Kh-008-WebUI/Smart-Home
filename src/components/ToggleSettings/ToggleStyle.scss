<<<<<<< HEAD
.settings_block {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: baseline;
  align-content: center;
  background-color: #44464f;
  color: white;
  width: 30%;
  height: 70px;
  border-radius: 7px;
}

.top_border {
  position: relative;
  top: 0;
  left: 0;
  background-color: #f6a821;
  width: 100%;
  height: 30%;
} 
   
=======
>>>>>>> origin
.switch {
  position: relative;
  top: 0;
  right: 0;
}
