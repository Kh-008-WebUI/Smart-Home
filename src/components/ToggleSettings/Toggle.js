import React from 'react';
import PropTypes from 'prop-types';
import './ToggleStyle.scss';

export default class ToggleSettings extends React.Component {

  constructor (props) {
    super(props);
    this.state = {
      value: props.checked
    };

    this.onChange = this.onChange.bind(this);
  }

  onChange () {
    this.setState({
      value: !this.state.value
    });
    this.props.sendToggleValue(!this.state.value);
  }

  render () {
    return (
      <div className={ `${ this.props.styleName }` }>
        <label className="switch">
          <input
            type="checkbox"
            checked={this.state.value}
            onClick={this.onChange}
          />
          <div className="slider round"></div>
        </label>
      </div>
    );
  }
}

ToggleSettings.propTypes = {
  styleName: PropTypes.string,
  checked: PropTypes.bool,
  sendToggleValue: PropTypes.func
};
