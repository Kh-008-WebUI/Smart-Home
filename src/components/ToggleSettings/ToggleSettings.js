import React from 'react';
import PropTypes from 'prop-types';
import './ToggleStyle.scss';

export default class ToggleSettings extends React.Component {

  constructor (props) {
    super(props);
  }

  onChangeValue = (e) => {
    this.props.setItemValue(!this.props.checked,
      this.props.itemId,
      this.props.deviceId);
  }

  render () {
    return (
      <div className={ `${ this.props.styleName }` }>
        <label className="switch">
          <input
            type="checkbox"
            checked={this.props.checked}
            onChange={this.onChangeValue}
          />
          <div className="slider round"></div>
        </label>
      </div>
    );
  }
}

ToggleSettings.propTypes = {
  device: PropTypes.object,
  styleName: PropTypes.string,
  checked: PropTypes.bool.isRequired,
  setItemValue: PropTypes.func,
  itemId: PropTypes.any,
  newValue: PropTypes.string,
  deviceId: PropTypes.string
};
ToggleSettings.defaultProps = {
  checked: false
};
