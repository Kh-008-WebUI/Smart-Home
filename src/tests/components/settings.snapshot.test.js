import React from 'react';
import ToggleSettings from '../../components/ToggleSettings/ToggleSettings';
import SelectLocation from '../../components/SelectLocation/SelectLocation';
import RangeSettings from '../../components/RangeSettings/Range';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const locations = [
  {
    _id:'59940beedfa2e10840ef5eac',
    value: 'kitchen',
    label: 'Kitchen',
  },
  {
    _id:'59940bf3dfa2e10840ef5ead',
    value: 'roof',
    label: 'Roof'
  },
  {
    _id: '599413dd9a6f1117dc4fe6a2',
    value: 'parking',
    label: 'Parking'
  }
];

describe('Settings UI test', () => {
  it('ToggleSettings', () => {
    const toggle = shallow(
      <ToggleSettings />
    );

    const rendered = toggle.find('input');

    expect(rendered.html().indexOf('checked={false}')).toEqual(-1);
  });

  it('SelectLocation', () => {
    const component = renderer.create(
      <SelectLocation
            selectLocation={jest.array}
            locations= {locations}
            addLocation={jest.fn()}
            deleteLocation={jest.fn()}
            defaultLocation='Kitchen'
            deviceExistInLocation={jest.fn()}
            deviceInLocation={jest.array}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('RangeSettings', () => {
    const component = renderer.create(
      <RangeSettings/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
