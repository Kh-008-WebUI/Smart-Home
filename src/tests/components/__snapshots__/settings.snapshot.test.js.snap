// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings UI test RangeSettings 1`] = `
<div
  className="undefined"
>
  <p
    className="range_value"
  >
    0
  </p>
  <input
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    step={1}
    type="range"
    value={0}
  />
</div>
`;

exports[`Settings UI test SelectLocation 1`] = `
<div>
  <div
    className="Select-control"
  >
    <div
      className="Select-value select-menu-container"
    >
      <div
        className="select-menu-label"
        onClick={[Function]}
      >
        <span
          className="Select-value-label"
        >
          Kitchen
        </span>
        <i
          className="select-toggle fa fa-caret-down"
        />
      </div>
    </div>
  </div>
  <div
    className="popup"
  >
    <p
      className="popup__header"
    >
      Confirm the action
    </p>
    <p
      className="popup__text"
    >
      Are you sure you want to delete this location?
    </p>
    <div
      className="popup__confirm"
    >
      <button
        className="btn popup__btn"
        disabled={undefined}
        onClick={[Function]}
        type="button"
      >
        Ok
      </button>
      <button
        className="btn btn--default popup__btn"
        disabled={undefined}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
